test/Transforms/multicontrol_decomposition.qke

```diff
@@ -39,6 +39,13 @@ func.func @cccx_negated_controls(%c0: !quake.ref,  %c1: !quake.ref, %c2: !quake.
   return
 }
 
+// CHECK-LABEL: func.func @c4x_negated_controls
+func.func @c4x_negated_controls(%c0: !quake.ref,  %c1: !quake.ref, %c2: !quake.ref, %c3: !quake.ref, %t: !quake.ref) {
+  // CHECK-NOT: quake.x [%{{.*}}, %{{.*}}, %{{.*}}, %{{.*}} neg [false, true, true, false]] %{{.*}}
+  quake.x [%c0, %c1, %c2, %c3 neg [false, true, true, false]] %t : (!quake.ref, !quake.ref, !quake.ref, !quake.ref, !quake.ref) -> ()
```

Comment:
Why not add the CHECK lines that verify the expected conversion is actually firing? As this is, it is merely checking that some transformation to eliminate the original op happened (and no clue which one) and that the unitary in and unitary out are equivalent. That is good of course but again doesn't really demonstrate that the actual transformation is being tested. It's like saying the final answer is `42`, but there are any number of equations that produce `42`.

Reply:
I don't think we need to commit checking for one particular way of doing multi control decomposition, we just need to guarantee that whatever way choose to do (and transform the circuit) will result in a equivalent circuit.

Reply:
Agree on not checking for a particular way of doing the decomposition. The question is: Do we also have a check that validates that there are no multi-controlled operations in the produced IR? if not, we should add that as well. The two tests together check both requirements, which are:
- eliminate all uses of multi-controlled
- do so in a manner that is correct (i.e. results in the same transformation of the quantum state).

Reply:
> Agree on not checking for a particular way of doing the decomposition. The question is: Do we also have a check that validates that there are no multi-controlled operations in the produced IR?

The pass itself don't generate operations with more than two controls. However, for us to make sure this won't happen (say if someone mistakenly change the pass), then it would require some sort of testing/verification pass as it might be complicated to check all edge cases using only FileCheck.

Maybe it is the case to create some generic verification pass that verifies certain IR properties depending on some pass options: e.g. `--verify-quake{max-controls=2}` (adding other options as we see fit)

Reply:
> As discussed, that seems reasonable to process in a separate PR. Please link the issue here to track that work, and then I'll approve this change.

See #311